src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php

```diff
@@ -167,4 +171,68 @@ protected function instantiateController($class)
     {
         return new $class();
     }
+
+    private function getControllerError($callable)
+    {
+        if (is_string($callable) && false !== strpos($callable, '::')) {
+            $callable = explode('::', $callable);
+        }
+
+        if (is_string($callable) && !function_exists($callable)) {
+            return sprintf('Function %s does not exist.', $callable);
+        }
+
+        if (is_array($callable)) {
```

Comment:
Can you test for `!is_array()` and return early? That's what we do in Symfony to avoid long condition bodies.